genai/context.py

```diff
@@ -77,32 +89,20 @@ def craft_output_message(output: Any) -> Dict[str, str]:
 ]
 
 
-def get_historical_context(
-    ipython: 'InteractiveShell', num_messages: int = 5, model: str = "gpt-3.5-turbo-0301"
-) -> List[Dict[str, str]]:
-    """Create a series of messages to use as context for ChatGPT."""
-    raw_inputs = ipython.history_manager.input_hist_raw
+def build_context(history_manager, start=1, stop=None):
+    context = Context()
 
-    # Now filter out any inputs that start with our filters
-    # This has to keep the input index as the key for the output
-    inputs = {}
-    for i, input in enumerate(raw_inputs):
-        if input is None or input.strip() == "":
+    for session, execution_counter, cell_text in history_manager.get_range(
+        session=0, start=start, stop=stop
+    ):
+        print('session', session)
+        if any(cell_text.startswith(token) for token in ignore_tokens):
             continue
 
-        if not any(input.startswith(token) for token in ignore_tokens):
-            inputs[i] = input
-
-    outputs = ipython.history_manager.output_hist
-    indices = sorted(inputs.keys())
-    context = []
-
-    # We will use the last `num_messages` inputs and outputs to establish context
-    for index in indices[-num_messages:]:
-        context.append(craft_user_message(inputs[index]))
+        context.append(cell_text, role="user", execution_count=execution_counter)
 
-        if index in outputs:
-            context.append(craft_output_message(outputs[index]))
+        output = history_manager.output_hist.get(execution_counter)
```

Comment:
TIL that there exists `history_manager.output_hist_reprs` that we _could_ use for the plain `repr`s so we don't have to compute them again.